_posts/2023-02-22-#25038.md

```diff
@@ -7,18 +7,234 @@ authors: [glozow]
+it will be evicted if its ancestor is replaced. However, a transaction that only conflicts with the
+descendant will be rejected due to it not signaling BIP125 replaceability *explicitly*.
+
+- Transactions must meet RBF fee rules on their own: Package RBF (see [definition](#Definitions)
+  above) is not currently possible. Package RBF would enable, for example, conflicting LN commitment
+transactions (which tend to have fees that are too similar to replace one another) to replace each
+other utilizing fees from a fee-bumping child.
+
+- Further reading on L2 pinning grievances:
+	* [Bringing a nuke to a knife fight: Transaction introspection to stop RBF pinning](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-May/020458.html)
+	* [RBF Pinning with Counterparties and Competing Interest](https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-April/002639.html)
+	* [Pinning : The Good, The Bad, The Ugly](https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-June/002758.html)
+	* [T-bast's Pinning Attacks Document](https://github.com/t-bast/lightning-docs/blob/master/pinning-attacks.md)
+	* [Eltoo Pinning](https://gist.github.com/instagibbs/60264606e181451e977e439a49f69fe1)
+
+A larger list of RBF limitations was discussed last year on [this mailing list
+post](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-January/019817.html) and
+[gist](https://gist.github.com/glozow/25d9662c52453bd08b4b4b1d3783b9ff). Various proposals to amend or rewrite RBF
+policy were discussed on those threads. The discussions resulted in this v3 proposal to address most issues. Additional
+RBF improvements (which do not conflict with v3) are also being considered.
+
+### V3 Policy
+
+The philosophy behind [v3](https://bitcoinops.org/en/topics/version-3-transaction-relay/) is that RBF limitations stem
+from current mempool policy being too permissive in the possible package topologies it accepts. [Package
+limits](https://github.com/bitcoin/bitcoin/blob/master/doc/policy/mempool-limits.md) exist to limit the worst-case
+computational complexity of mempool operations such as block template assembly and new transaction validation. However,
+current limits still allow huge clusters of thousands of transactions.
+
+<!-- FIXME: add photo -->
+
+With these possible transaction graphs, it isn't feasible to add a comparison of incentive compatibility scores, as
+calculating them could involve iterating through the entire mempool. Permissive package limits are also what makes Rule
+3 pinning possible; when two counterparties share a transaction, they are both able to bring in descendants and other
+transactions to influence the incentive compatibility of their shared transaction.
+
+On the other hand, most protocols involving untrusted counterparties don't require complex transaction graphs. Usually,
+a 1-parent-1-child package is sufficient: the presigned transaction plus a child attaching fees at broadcast time.
```

Comment:
I think this needs more highlighting that a core principle of V3 is about _reducing_ options: radically limit the topology and complexity of packages/descendants so that it becomes much easier to reason about all possible scenarios and people _can_ use it if they want to

Reply:
"The philosophy behind v3 is that RBF limitations stem from current mempool policy being too permissive in the possible package topologies it accepts." ?